src/modules/UI/scenes/Plugins/EdgeProvider.js

```diff
@@ -420,55 +423,64 @@ export class EdgeProvider extends Bridgeable {
+      if (response.status !== 201) {
+        const errorData = await response.json()
+        throw new Error(errorData.errors[0].code + ' ' + errorData.errors[0].message)
       }
-      const signedTransactionResponse = await window.fetch(newURL, request)
-      LOGGER.log('Bity signedTransaction: ', JSON.stringify(signedTransaction))
-      LOGGER.log('Bity transaction message body: ', JSON.stringify(body))
-      LOGGER.log('Bity signedTransactionResponse: ', JSON.stringify(signedTransactionResponse))
-      if (signedTransactionResponse.status === 400) {
-        throw new Error('Could not complete transaction. Code: 470')
+      const newURL = url2 + response.headers.get('Location')
+      console.log('Bity newUrl: ', newURL)
+      const request2 = {
+        method: 'GET',
+        credentials: 'include'
       }
-      if (signedTransactionResponse.status === 204) {
-        const bankDetailsRequest = {
-          method: 'GET',
-          credentials: 'include'
+      const response2 = await window.fetch(newURL, request2)
+      console.log('Bity response2: ', response2)
+      if (response2.status !== 200) {
+        throw new Error('Problem confirming order: Code n200')
+      }
+      const orderData = await response2.json()
+      console.log('Bity orderData: ', orderData)
+      if (orderData.message_to_sign) {
+        const { signature_submission_url, body } = orderData.message_to_sign
+        const signedTransaction = await this.signMessage(body)
+        const newURL = url2 + signature_submission_url
+        const request = {
+          method: 'POST',
+          headers: {
+            Host: 'exchange.api.bity.com',
+            'Content-Type': '*/*'
+          },
+          body: signedTransaction
+        }
+        const signedTransactionResponse = await window.fetch(newURL, request)
+        console.log('Bity signedTransactionResponse: ', signedTransactionResponse)
+        if (signedTransactionResponse.status === 400) {
+          throw new Error('Could not complete transaction. Code: 470')
         }
-        const detailUrl = url + '/' + orderData.id
-        const bankDetailResponse = await window.fetch(detailUrl, bankDetailsRequest)
-        if (bankDetailResponse.status === 200) {
-          const parsedResponse = await bankDetailResponse.json()
-          return parsedResponse
+        if (signedTransactionResponse.status === 204) {
+          const bankDetailsRequest = {
+            method: 'GET',
+            credentials: 'include'
+          }
+          const detailUrl = url + '/' + orderData.id
+          const bankDetailResponse = await window.fetch(detailUrl, bankDetailsRequest)
+          if (bankDetailResponse.status === 200) {
+            const parsedResponse = await bankDetailResponse.json()
+            console.log('Bity parsedResponse: ', parsedResponse)
+            return parsedResponse
+          }
         }
       }
+      return orderData
+    } catch (e) {
```

Comment:
When I mentioned to add a catch, it was to add a catch to each of the fetch calls individually so that you know which one actually threw. It would be easiest to add the .catch() notation after each fetch call.

```
await window.fetch(newURL, request).catch(e => {
  console.log(`throw from fetch ${newURL}`, e)
  throw e
})```